runtime/doc/lua-intro.txt

```diff
@@ -0,0 +1,107 @@
+*lua-intro.txt*	Nvim
+
+		       NVIM REFERENCE MANUAL
+
+			Introduction to Lua
+
+
+This chapter will give you the basics of using Lua in Neovim.
+
+                                      Type |gO| to see the table of contents.
+
+==============================================================================
+Introduction							     *lua-intro*
```

Comment:
I think the idea is that this is a full, practical "guide". Suggest we call it `lua-guide` or `lua-usage` (open to better ideas). It's much more than an intro, but not really a tutorial.